flight/Modules/Battery/battery.c

```diff
@@ -149,14 +152,21 @@ static void batteryTask(void * parameters)
 					scaled_voltage = 0.0f;
 					adc_offset_invalid = true;
 				}
+				else if(scaled_voltage > 3.0f && !cells_calculated && (batterySettings.MaxCellVoltage > 0)){
+					cells_calculated = true;
+					cells = ((scaled_voltage / batterySettings.MaxCellVoltage) + 1);
+					flightBatteryData.DetectedCellCount = cells;
```

Comment:
`FlightBatterySettingsNbCellsSet((uint8_t*)&cells);`
Could be used too instead of the new var. in batterystate but should be avoided.